Libraries/Blob/File.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule File
+ * @flow
+ */
+'use strict';
+
+const Blob = require('Blob');
+
+/**
+ * The File interface provides information about files.
+ */
+class File extends Blob {
+  /**
+   * Name of the file.
+   */
+  get name(): string {
+    return this.data.name || '';
```

Comment:
I believe it is impossible to have a file without name. Should we generate something like uuid for each blob w/o name?

Reply:
It's valid, you can create a file object without name with `new File([""], "");`

Reply:
Chrome throws if you don't pass a the parts and name args so I added an invariant. Behavior is slightly different since chrome accepts `new File([], undefined)` and stringifies it but I think it's better to not allow that. Still need the `|| ''` to make flow happy though.

Reply:
Let's add an invariant that `this.data.name` is a string instead of defaulting to `''` (if it's just to make Flow happy). If it is anything else, then some constraint was violated between the constructor and invocation of this method.